packages/mobile/src/store/products/actions.ts

```diff
@@ -13,4 +17,74 @@ const loadProducts = createAsyncThunk<
   return productApi.getProducts(requestParams);
 });
 
-export { loadProducts };
+const fetchFavorites = createAsyncThunk<
+  ProductDto[],
+  ProductRequestDto,
+  AsyncThunkConfig
+>(ActionType.FETCH_FAVORITES, async (requestParams, { extra }) => {
+  const { productApi } = extra;
+
+  return await productApi.getFavorites(requestParams);
+});
+
+const fetchFavoriteIds = createAsyncThunk<
+  string[],
+  undefined,
+  AsyncThunkConfig
+>(ActionType.FETCH_FAVORITES_IDS, async (_, { extra }) => {
+  const { productApi } = extra;
+
+  return await productApi.getFavoritesIds();
+});
+
+const addToFavorite = createAsyncThunk<
+  ProductIdRequest,
+  string,
+  AsyncThunkConfig
+>(ActionType.ADD_TO_FAVORITE, async (productId, { extra }) => {
+  const { productApi } = extra;
+
+  return await productApi.uploadToFavorites({ productId });
+});
+
+const deleteFromFavorite = createAsyncThunk<
+  ProductIdRequest,
+  string,
+  AsyncThunkConfig
+>(ActionType.DELETE_FROM_FAVORITE, async (productId, { extra }) => {
+  const { productApi } = extra;
+
+  return await productApi.deleteFromFavorites({ productId });
+});
+
+const addToFavoriteGuestUser = createAction<PrepareAction<string>, string>(
+  ActionType.ADD_TO_FAVORITE,
+  (productId) => {
+    return { payload: productId };
+  },
+);
+
+const deleteFromFavoriteGuestUser = createAction<PrepareAction<string>, string>(
+  ActionType.DELETE_FROM_FAVORITE,
+  (productId) => {
+    return { payload: productId };
+  },
```

Comment:
No need to create separate actions. You may determine if the user is logged in by `userId`